### PR TITLE
chore: set TypeScript version range to >=4.2

### LINKED ISSUE #68: Determine TypeScript version support
### Overview

What version range of TypeScript should this plugin support?

#### Actual Behavior

Apparently it goes back to 2.0:

https://github.com/JoshuaKGoldberg/eslint-plugin-expect-type/blob/312aa1e768472b2546d0cd3786d2e5ff01ec74f5/src/rules/expect.ts#L611

#### Expected Behavior

I'd propose going along with https://github.com/DefinitelyTyped/DefinitelyTyped/#support-window: versions of TypeScript less than 2 years old.

Let's:

* Explicitly document this in the README.md
* Remove that TypeScript support shim

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing issue: fixes #68
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/eslint-plugin-expect-type/labels/status%3A%20accepting%20prs)

## Overview

Sets the minimum tested TypeScript version to 4.2, with a matrix of tests across versions up through `5.0` and `latest`.

Borrows some GHA configuration from https://github.com/JoshuaKGoldberg/template-typescript-node-package.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,9 +44,9 @@ Then, you'll be able to use `^?`, `$ExpectError`, `$ExpectType`, and `$ExpectTyp
 🔧 Automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/user-guide/command-line-interface#--fix).\
 💭 Requires type information.
 
-| Name                           | Description                                | 💼 | 🔧 | 💭 |
-| :----------------------------- | :----------------------------------------- | :- | :- | :- |
-| [expect](docs/rules/expect.md) | Expects type error, type snapshot or type. | ✅  | 🔧 | 💭 |
```
Comment (JoshuaKGoldberg) [inline README.md:49]: 🤷 the doc generator wanted it.

File: .github/actions/prepare/action.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+description: Prepares the repo for a typical CI job
+
+name: Prepare
+runs:
+  steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-node@v3
+      with:
+        cache: 'npm'
```
Comment (danvk) [inline .github/actions/prepare/action.yml:9]: nit: this value doesn't need to be quoted in YAML (maybe the 18, too).

File: .github/workflows/format.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/prepare
+      - run: npm run format:check
+
+name: Format
+
+on:
+  pull_request: ~
```
Comment (danvk) [inline .github/workflows/format.yml:12]: TIL that `~` means `null` in YAML. Perhaps just write it as `null` for clarity?

Is this saying that we don't want to run format checks on PR branches? Why would that be?

**Review by danvk — APPROVED** [review]
Thanks @JoshuaKGoldberg!

File: .github/workflows/format.yml (hunk 3 above)
Comment (JoshuaKGoldberg) [reply .github/workflows/format.yml:12]: Heh I went the opposite conclusion in https://github.com/JoshuaKGoldberg/template-typescript-node-package - that it implies _"run on all of them"_, which `null` feels ...  the opposite to me?

It means we run format checks on:
* All pull request commits
* Commits to `main`

**Pull request merged**